src/__test__/useConnection.test.ts

```diff
@@ -0,0 +1,86 @@
+import { renderHook, act } from "@testing-library/react-hooks";
+import Onboard from "bnc-onboard";
+import * as ethers from "ethers";
+import {
+  API as OnboardApi,
+  Initialization,
+} from "bnc-onboard/dist/src/interfaces";
+
+import { useConnection, ConnectionProvider } from "../hooks";
+
+function deferred() {
+  let resolve, reject;
+  const promise = new Promise((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return { promise, resolve, reject };
+}
+
+jest.mock("bnc-onboard");
+const mockedOnboard = Onboard as jest.Mock<Partial<OnboardApi>>;
+const mockedWalletSelect = jest.fn();
+const mockedWalletCheck = jest.fn();
+
+const mockProviderFn: ethers.providers.JsonRpcFetchFunc = jest.fn();
+
+jest
+  .spyOn(ethers.providers.Web3Provider.prototype, "getNetwork")
+  .mockImplementation(async () => {
+    return {
+      chainId: 1,
+      name: "mainnet",
+    };
+  });
+let fakeProvider: ethers.providers.Web3Provider;
+let fakeNetwork: ethers.providers.Network;
+let fakeAddress: string;
+
+beforeAll(async () => {
+  fakeProvider = new ethers.providers.Web3Provider(mockProviderFn);
+  fakeNetwork = await fakeProvider.getNetwork();
+});
+
+test("should connect to the provider and save the data", async () => {
+  const { promise, resolve } = deferred();
+
+  mockedOnboard.mockImplementation(({ subscriptions }: Initialization) => {
+    promise.then(() => {
```

Comment:
we talked about doing this as an event emitter instead. will leave it up to you though if you want to change it.